SplitFiles/control.c

```diff
@@ -48,20 +48,29 @@ char* buffer_getcommand(const char c) {
 }
 
 //show the stack on the screen
-void buffer_restack(const ControlBuffer* const buff) {
+void buffer_restack(ControlBuffer* buff) {
     int i;
     char* command;
-    const int buffer_index = buff->bufferIndex;
+    const int bufferIndex = buff->bufferIndex;
+    int prevBufferIndex = buff->prevBufferIndex;
     // Write out commands
-    printf(PRINTAT"%c%c""%-12s", 18, 12, "Commands");
-    for (i = 0; i < buffer_index; ++i) {
-        command = buffer_getcommand((buff->buffer[i]));
-        printf(PRINTAT"%c%c""%-12s", 18, i + 13, command);
+    if (prevBufferIndex < 0) { //Do this once
+        printf(PRINTAT"%c%c""%-12s", 18, 12, "Commands");
```

Comment:
Don't need to print this every time as it's at a fixed position (I have addressed this in my branch but you can just remove it here if you like